substratest/factory.py

```diff
@@ -276,26 +311,69 @@ def _get_keys(obj, field='key'):
 
 @dataclasses.dataclass
 class ComputePlanSpec(_Spec):
-    algo_key: str
     objective_key: str
     traintuples: typing.List[ComputePlanTraintupleSpec]
+    composite_traintuples: typing.List[ComputePlanCompositeTraintupleSpec]
+    aggregatetuples: typing.List[ComputePlanAggregatetupleSpec]
     testtuples: typing.List[ComputePlanTesttupleSpec]
 
-    def add_traintuple(self, dataset, data_samples, traintuple_specs=None, tag=None):
-        traintuple_specs = traintuple_specs or []
+    def add_traintuple(self, algo, dataset, data_samples, in_models_tuples=None, tag=''):
```

Comment:
IMHO, `tuple_specs` describes better the input variable than `in_models_tuples` (even if it's not aligned with the substra library).

Reply:
I understand what you mean but I find `tuple_specs` not clear enough (not clear that it is supposed to mach the in_models and not clear that we're talking about *ComputePlanSpecs here). 
What about `in_tuple_cp_specs`?

Reply:
I'm not really convinced with `in_tuple_cp_specs`: `cp` is not explicit and I find it too verbose. 

In your initial proposition, I think that `models` and `tuples` are redundant (as a model is the output of a tuple). What about `in_models` then? It would be consistent with the substra library even if I don't think it is the best name as it **is** a tuple.

Reply:
I agree it's not the best name, but at least it's consistent. This is something that ought to be changed throughout the platform. Let's go with in_models!